arrow-fx-reactor/src/main/kotlin/arrow/fx/reactor/extensions/ForwardDisposable.kt

```diff
@@ -0,0 +1,70 @@
+package arrow.fx.reactor.extensions
+
+import arrow.fx.internal.Platform
+import kotlinx.atomicfu.atomic
+import reactor.core.Disposable
+
+/**
+ * [ForwardDisposable] allows you to link 2 [Disposable]s.
+ *
+ * This is useful for when you want to link the cancel lifecycle of an `Emitter`
+ * to running an [Disposable] stream.
+ *
+ * This is used in `bracket`, `guaranteeCase` & `asyncF`.
+ */
+internal class ForwardDisposable {
+
+  private val state = atomic(init)
+
+  fun cancel(): Disposable {
```

Comment:
It's actually a bit trickier than that 😅
This doesn't really cancel anything. `ForwardDisposable` exposes 2 functions.

- `cancel`, which returns a `Disposable` that can be linked to an `Sink` using `onCancel`.
- `complete`, to atomically set the `Disposable` of the stream that is running within the coupled `Sink`.

I kept this the same as for `IO` since it's only used internally and not exposed.